src/internal/mod.rs

```diff
@@ -24,7 +24,7 @@ use std::{
 
 pub mod document_api;
 pub mod group_api;
-mod rest;
+pub(crate) mod rest; //COLT: Feels a bit bad.
```

Comment:
an alternative would be to allow the internal API to generate the `IronCoreRequest` struct, allowing it to stay private to internal.